google/cloud/security/common/gcp_api/admin_directory.py

```diff
@@ -132,16 +125,11 @@ def get_groups(self, customer_id='my_customer'):
         """
         groups_stub = self.service.groups()
         request = groups_stub.list(customer=customer_id)
-        results_by_group = []
-
-        # TODO: Investigate yielding results to handle large group lists.
-        while request is not None:
-            try:
-                with self.rate_limiter:
-                    response = self._execute(request)
-                    results_by_group.extend(response.get('groups', []))
-                    request = groups_stub.list_next(request, response)
-            except (HttpError, HttpLib2Error) as e:
-                raise api_errors.ApiExecutionError(groups_stub, e)
-
-        return results_by_group
+
+        try:
+            results = self._build_paged_result(
+                request, groups_stub, self.rate_limiter)
+        except api_errors.UnsupportedApiError:
```

Comment:
See my comment above.  Also, by doing this here, I don't think that a specific message about unsupported api would surface.